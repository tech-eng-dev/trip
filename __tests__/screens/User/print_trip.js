import React from 'react';
import { shallow, configure } from 'enzyme';
import configureStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';
// imported as a connected component!
import UserPrintTrip from '../../../src/screens/User/print_trip';

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);
configure({ adapter: new Adapter() });

const initialState = {
  myRecords: {
    1518174116204: {
      comment: 'Test',
      createdTime: '1518174116204',
      startDate: '2018-02-04',
      endDate: '2018-02-28',
      location: {
        description: 'test',
        geometry: {
          location: {
            lat: '12.34',
            lng: '12.34',
          },
          viewport: {
            northeast: {
              lat: '12.34',
              lng: '12.34',
            },
            southwest: {
              lat: '12.34',
              lng: '12.34',
            },
          },
        },
        id: 'ad013b06650801c352d643dc90db93210e5d07cb',
      },
    },
  },
};

describe('UserPrintTrip component renders correctly', () => {
  it('renders as expected', () => {
    const store = mockStore(initialState);
    const wrapper = shallow(
      <UserPrintTrip />,
      { context: { store } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
