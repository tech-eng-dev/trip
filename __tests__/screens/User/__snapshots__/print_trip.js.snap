// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPrintTrip component renders correctly renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserPrintTrip
    fetchAllRecords={[Function]}
    fetchBlockData={[Function]}
    fetchMyRecords={[Function]}
    fetchUsers={[Function]}
    listenUserProfile={[Function]}
    logout={[Function]}
    myRecords={
      Object {
        "1518174116204": Object {
          "comment": "Test",
          "createdTime": "1518174116204",
          "endDate": "2018-02-28",
          "location": Object {
            "description": "test",
            "geometry": Object {
              "location": Object {
                "lat": "12.34",
                "lng": "12.34",
              },
              "viewport": Object {
                "northeast": Object {
                  "lat": "12.34",
                  "lng": "12.34",
                },
                "southwest": Object {
                  "lat": "12.34",
                  "lng": "12.34",
                },
              },
            },
            "id": "ad013b06650801c352d643dc90db93210e5d07cb",
          },
          "startDate": "2018-02-04",
        },
      }
    }
    saveAllRecords={[Function]}
    saveBlockData={[Function]}
    saveMyRecords={[Function]}
    saveUserPhoto={[Function]}
    saveUserProfile={[Function]}
    saveUsers={[Function]}
    setFilterOption={[Function]}
    showToast={[Function]}
    welcome={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <CustomHeader
          backgroundColor="#3DA0F1"
          left="ios-arrow-back"
          onPressLeft={[Function]}
          onPressRight={[Function]}
          right={undefined}
          title="Print"
        />,
        <UNDEFINED
          contentContainerStyle={
            Object {
              "alignItems": "center",
              "paddingHorizontal": 15,
              "paddingVertical": 40,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#5C6063",
                "padding": 10,
              }
            }
          >
            You will get printed Trips for next month.
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#5C6063",
                "padding": 10,
              }
            }
          >
            But you have no travel plans next month.
          </Text>
        </UNDEFINED>,
      ],
      "style": Object {
        "backgroundColor": "#F5FAFE",
        "flex": 1,
      },
      "testID": "print-root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backgroundColor": "#3DA0F1",
          "left": "ios-arrow-back",
          "onPressLeft": [Function],
          "onPressRight": [Function],
          "right": undefined,
          "title": "Print",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#5C6063",
                  "padding": 10,
                }
              }
            >
              You will get printed Trips for next month.
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#5C6063",
                  "padding": 10,
                }
              }
            >
              But you have no travel plans next month.
            </Text>,
            null,
          ],
          "contentContainerStyle": Object {
            "alignItems": "center",
            "paddingHorizontal": 15,
            "paddingVertical": 40,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "You will get printed Trips for next month.",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#5C6063",
                "padding": 10,
              },
            },
            "ref": null,
            "rendered": "You will get printed Trips for next month.",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "But you have no travel plans next month.",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#5C6063",
                "padding": 10,
              },
            },
            "ref": null,
            "rendered": "But you have no travel plans next month.",
            "type": [Function],
          },
          null,
        ],
        "type": undefined,
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <CustomHeader
            backgroundColor="#3DA0F1"
            left="ios-arrow-back"
            onPressLeft={[Function]}
            onPressRight={[Function]}
            right={undefined}
            title="Print"
          />,
          <UNDEFINED
            contentContainerStyle={
              Object {
                "alignItems": "center",
                "paddingHorizontal": 15,
                "paddingVertical": 40,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#5C6063",
                  "padding": 10,
                }
              }
            >
              You will get printed Trips for next month.
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#5C6063",
                  "padding": 10,
                }
              }
            >
              But you have no travel plans next month.
            </Text>
          </UNDEFINED>,
        ],
        "style": Object {
          "backgroundColor": "#F5FAFE",
          "flex": 1,
        },
        "testID": "print-root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backgroundColor": "#3DA0F1",
            "left": "ios-arrow-back",
            "onPressLeft": [Function],
            "onPressRight": [Function],
            "right": undefined,
            "title": "Print",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#5C6063",
                    "padding": 10,
                  }
                }
              >
                You will get printed Trips for next month.
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#5C6063",
                    "padding": 10,
                  }
                }
              >
                But you have no travel plans next month.
              </Text>,
              null,
            ],
            "contentContainerStyle": Object {
              "alignItems": "center",
              "paddingHorizontal": 15,
              "paddingVertical": 40,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "You will get printed Trips for next month.",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#5C6063",
                  "padding": 10,
                },
              },
              "ref": null,
              "rendered": "You will get printed Trips for next month.",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "But you have no travel plans next month.",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#5C6063",
                  "padding": 10,
                },
              },
              "ref": null,
              "rendered": "But you have no travel plans next month.",
              "type": [Function],
            },
            null,
          ],
          "type": undefined,
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
